openedx/core/djangoapps/credit/signals.py

```diff
@@ -33,25 +32,6 @@ def on_course_publish(course_key):
         log.info(u'Added task to update credit requirements for course "%s" to the task queue', course_key)
 
 
-@receiver(SignalHandler.pre_publish)
-def on_pre_publish(sender, course_key, **kwargs):  # pylint: disable=unused-argument
-    """
-    Create user partitions for verification checkpoints.
-
-    This is a pre-publish step since we need to write to the course descriptor.
-    """
-    from openedx.core.djangoapps.credit import api
-    if api.is_credit_course(course_key):
```

Comment:
We haven't refactored all of our grades tests to separate out the unit tests from the integration tests.  The query count you're referring to is an integration-like test - it's good to know how many queries, in total, to expect whenever a grade is calculated - including downstream queries.
Regarding this credit code, I wonder if this app is still needed.  Don't certificates supercede this?